nailgun-server/src/main/java/com/martiansoftware/nailgun/NGSecurityManager.java

```diff
@@ -58,10 +60,144 @@ public void checkPermission(Permission perm, Object context) {
     }
   }
 
-  /** Avoid constructing a FilePermission object in checkRead if base manager is null. */
+  // Overrides below avoid the cost of creating Permissions objects if base manager is null.
+  // FilePermission, in particular, is expensive to create. 
+
   public void checkRead(String file) {
     if (base != null) {
       super.checkRead(file);
```

Comment:
Should this be `base.checkRead()` instead of `super.checkRead()`? I have the same question for the new overloads added in this PR. It looks like @qma asked the same question on the original patch: https://github.com/facebook/nailgun/pull/11#commitcomment-13058970

Reply:
On reflection, I suspect that this was done for backwards-compatibility reasons, since overriding and delegating to `base` would be a behavior change whereas this change is only a performance optimization.

Reply:
`super.checkRead` will eventually get to `base.checkPermission`. But it would be potentially more efficient to call `base.checkRead` directly here. The semantics should be the same for any well behaved `SecurityManager`.

Reply:
I've pushed a commit that delegates directly to the corresponding method.